deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/params/DefaultParamInitializer.java

```diff
@@ -162,6 +181,20 @@ protected INDArray createBias(long nOut, double biasInit, INDArray biasParamView
         return biasParamView;
     }
 
+    protected INDArray createGain(NeuralNetConfiguration conf, INDArray gainParamView, boolean initializeParameters) {
+        org.deeplearning4j.nn.conf.layers.FeedForwardLayer layerConf =
+                (org.deeplearning4j.nn.conf.layers.FeedForwardLayer) conf.getLayer();
+        return createGain(layerConf.getNOut(), layerConf.getGainInit(), gainParamView, initializeParameters);
+    }
+
+    protected INDArray createGain(long nOut, double gainInit, INDArray gainParamView, boolean initializeParameters) {
+        if (initializeParameters) {
+            INDArray ret = Nd4j.valueArrayOf(new long[] {1, nOut}, gainInit);
+            gainParamView.assign(ret);
```

Comment:
Don't create this temporary ret array. Use ```gainParamView.assign(gainInit)``` instead.

Reply:
good idea, I'll change it accordingly in the createBias method as well then.